### PR TITLE
Split limitrange tests

### LINKED ISSUE #78828: Promote LimitRange Tests to Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the LimitRange kind:

[replaceCoreV1NamespacedLimitRange](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-replaceCoreV1NamespacedLimitRange&showUntested=false&showConformanceTested=false&level=stable)

[deleteCoreV1NamespacedLimitRange](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-deleteCoreV1NamespacedLimitRange&showUntested=false&showConformanceTested=false&level=stable)

[createCoreV1NamespacedLimitRange](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-createCoreV1NamespacedLimitRange&showUntested=false&showConformanceTested=false&level=stable)

[readCoreV1NamespacedLimitRange](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-readCoreV1NamespacedLimitRange&showUntested=false&showConformanceTested=false&level=stable)

These are stable core endpoints, which have existing testing, potentially making for low-hanging fruit to upgrade to Conformance.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Splits [LImitRange test](https://github.com/kubernetes/kubernetes/pull/78921) into multiple tests.

**Which issue(s) this PR fixes**:
Fixes #78828

**Special notes for your reviewer**:
Part of umbrella issue #78748

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area test

@kubernetes/sig-architecture-pr-reviews
@kubernetes/sig-scheduling-pr-reviews

### CODE REVIEW

File: test/e2e/scheduling/limit_range.go [hunk 1]
```diff
@@ -44,97 +44,34 @@ const (
...
-		limitRange, err = f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Get(limitRange.Name, metav1.GetOptions{})
+		limitRange, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Get(limitRange.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err)
-		expected := v1.ResourceRequirements{Requests: defaultRequest, Limits: defaultLimit}
 		actual := v1.ResourceRequirements{Requests: limitRange.Spec.Limits[0].DefaultRequest, Limits: limitRange.Spec.Limits[0].Default}
 		err = equalResourceRequirement(expected, actual)
 		framework.ExpectNoError(err)
+	})
 
+	ginkgo.It("should create a Pod with no resource requirements", func() {
```
Comment (timothysc) [inline test/e2e/scheduling/limit_range.go]: So I disagree that we should be breaking up this function into multiple It() blocks.  

It(should be the behavior you are verifying in the test)
By (whatever means necessary) 

^ It is ok to have multiple By()'s for a given test, but it's not ok to have multiple It()'s in a test if they are not germain to the the intent or subpackage.

**Review by timothysc — COMMENTED** [review]
/hold

File: test/e2e/scheduling/limit_range.go (hunk 1 above)
Comment (devaips) [reply test/e2e/scheduling/limit_range.go]: @timothysc Did you see @spiffxp's [comment about splitting this test into multiple tests](https://github.com/kubernetes/kubernetes/pull/78921#discussion_r295536235) due how busy it is?

Comment (alejandrox1) [reply test/e2e/scheduling/limit_range.go]: friendly ping @timothysc

Comment (alejandrox1) [reply test/e2e/scheduling/limit_range.go]: Actually, just noticed this other comment https://github.com/kubernetes/kubernetes/pull/78921#discussion_r298771876

Comment (timothysc) [reply test/e2e/scheduling/limit_range.go]: So you can break apart the test by cleaning up the behavior into clean functions and possibly collapsing the By's.  Comment above still holds.  Multiple It(s) is not the right approach.

**Comment by alejandrox1** [comment]
skipped my mind that this is part of work by conformance
/sig architecture
/area conformance

**Comment by ttousai** [comment]
Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on Nov. 18. Is this PR still intended for 1.17?

**Comment by BobyMCbobs** [comment]
> Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on Nov. 18. Is this PR still intended for 1.17?

I'd like to get it in, but the next release is fine if this one doesn't work out.

**Comment by ttousai** [comment]
@BobyMCbobs please address https://github.com/kubernetes/kubernetes/pull/81311#discussion_r331593029

**Comment by spiffxp** [comment]
/close
Sorry for the back and forth, I think I now agree with @timothysc's comment.  Mechanically turning By's into It's wasn't quite my intent, but doing so semantically is just too fussy.  The test is well written as-is, and I would accept the test for promotion as was tried in https://github.com/kubernetes/kubernetes/pull/78921

Please open a new PR for promotion and we'll get it through

**Comment by hh** [comment]
@BobyMCbobs please write a promotion PR

**Pull request closed without merging**